### PR TITLE
Bug 1953518: cmd/thanos: use miekgdns resolver as default

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

File: cmd/thanos/rule.go [hunk 1]
```diff
@@ -120,8 +120,11 @@ func registerRule(app *extkingpin.App) {
 	httpMethod := cmd.Flag("query.http-method", "HTTP method to use when sending queries. Possible options: [GET, POST]").
 		Default("POST").Enum("GET", "POST")
 
+	// The native Go resolver fails to parse compressed responses for SRV records.
+	// The miekgdns resolver doesn't suffer the same issue hence defaulting to it instead.
+	// See https://bugzilla.redhat.com/show_bug.cgi?id=1953518
 	dnsSDResolver := cmd.Flag("query.sd-dns-resolver", "Resolver to use. Possible options: [golang, miekgdns]").
```
Comment (paulfantom) [inline cmd/thanos/rule.go:126]: Why not pass it as flag in https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/thanos-ruler.libsonnet#L376-L393 ? We already have this pattern for querier in https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/thanos-querier.libsonnet#L345

Comment (simonpasquier) [reply cmd/thanos/rule.go:126]: we don't have a way to pass the flag with the Prometheus operator (at least I've tried and haven't found).

Comment (paulfantom) [reply cmd/thanos/rule.go:126]: ack.

/lgtm

**Comment by simonpasquier** [comment]
/hold
I'd like to validate the change with a local cluster

**Comment by simonpasquier** [comment]
/hold cancel
Verified with a test cluster that the error goes away...

**Pull request merged**